test/Microsoft.Extensions.Configuration.FileProviderExtensions.Test/ConfigurationRootExtensionsTest.cs

```diff
@@ -40,12 +40,59 @@ public void ReloadOnChanged_GetTokenBeforeReload()
             Assert.Equal(1, configuration.ReloadCount);
         }
 
+        [Fact]
```

Comment:
@rynowak Please confirm that the following 3 behavior verification tests align with your expectations :)